### ISSUE TITLE
isRecordType in SObject behaves differently for default type and when a record type is set

### ISSUE THREAD

**Issue by halderanindya** [SalesforceLabs/FluentAssert#14]
Hi 
I was trying to use isRecordType from SObject and I found this issue - 

I created account like this - 
Id customerRecTypeId = Schema.SObjectType.Account.getRecordTypeInfosByName().get('Customer').getRecordTypeId();
    Account acc = new Account(
        Name = 'TEST ACCOUNT',
        Source_ID__c = 'ACCOUNT 001',
        recordTypeId = customerRecTypeId
    );

insert ACC;

Fluent.Assert.that(acc).isRecordType('Customer') -> This fails
But
Fluent.Assert.that(acc).isRecordType(customerRecTypeId) -> This passes.

So it looks like if a record type is set during record creation then it compares against record type id not record type developer name.

I also looked into your code
```
global SObjectAssert isRecordType(String developerName) {
        notNull(actual, 'actual');
        notNull(developerName, 'developerName');

        **String actualRecordTypeDevName = actual.isSet('RecordTypeId') ? (String) actual.get('RecordTypeId') :** SObjectTypeUtil.getDefaultRecordTypeInfo(Schema.SObjectType.Account).getDeveloperName();
        assert(developerName == actualRecordTypeDevName, 'Expected RecordType to be \'{0}\' but found \'{0}\'', new List<Object> {developerName, actualRecordTypeDevName});

        return this;
    } 
```
It looks like if record type id is set then you are assigning that id to actualRecordTypeDevName . I was expecting you would fetch the developer name from that record type id and use that to compare the values.

Is this an expected behaviour ?

**Comment by noerremark** [comment]
@halderanindya - Sorry that I didn't notice you created this issue. Let me investigate and I'll get back to you, but to me, it seems your right in your assumptions.

**Comment by noerremark** [comment]
@halderanindya - Thanks for your patience. Can you review this PR (#15)  and the tests to verify that it fixes the issue? Properly testingRecordTypes requires a bit more than just a package, so I added some sample meta-data for verification.
Once reviewed, I can pretty quickly create a package for it.

**Comment by noerremark** [comment]
@halderanindya - Thanks for the review! The issue is fixed in 1.3.0, which has Id `04t5f000000JkGsAAK`. Feel free to report any other issues / feature requests that you may have.
